drv/lpc55-sprot-server/src/main.rs

```diff
@@ -0,0 +1,426 @@
+                        let b = self.spi.read_u8();
+                        let incr = if self.rxcount < rx_end {
+                            io = true;
+                            self.rx[self.rxcount] = b;
+                            1
+                        } else {
+                            0
+                        };
+                        self.rxcount += incr;
+                    }
+                    if !io {
+                        break;
+                    }
+                }
+                if !inframe {
+                    // If CSn was deasserted, then the IO loop just completed
+                    // would have fetched the remaining bytes out of FIFORD
+                    // and our work is done.
+                    // We're sending out trailing bytes in FIFOWR to avoid
+                    // Tx underrun errors when we get more Rx bytes than we
+                    // are sending. So, expect that there are always some bytes
+                    // remaining in FIFOWR.
+                    // Actual transmitted bytes are always going to equal the
+                    // number of received bytes. Since the rx_count is not
+                    // scewed by the Tx trailing bytes still in FIFOWR, just
+                    // use rx_count for both.
+
+                    // let fifostat = self.spi.fifostat();
+
+                    // Update Tx state and account for trailing bytes
+                    // left in FIFOWR if present.
+                    self.txremainder = fifostat.txlvl().bits() as usize;
+                    break 'outer;
+                }
+            } // FIFO polling loop
+        }
+        // Done, deassert ROT_IRQ
+
+        // XXX Denial of service by forever asserting CSn?
+        // We could mitigate by imposing a time limit
+        // and resetting the SP if it is exceeded.
+        // But, the management plane is going to notice that
+        // the RoT is not available. So, does it matter?
+
+        // Any data remaining in FIFORD that could be comsumed,
+        // has been consumed. So, close out the received message.
+
+        // The following drain should be redundant with the one at the
+        // beginning of this function.
+        // However, if SP sends/receives while we're away processing
+        // a message, then we will see an underrun, and possibly an
+        // overrun, when we come back instead of just a Tx not full interrupt.
+        self.spi.drain_tx();
+        self.spi.send_u8(VERSION_BUSY);
+        self.gpio.set_val(ROT_IRQ, Value::One).unwrap_lite();
+    }
+}
+
+fn turn_on_flexcomm(syscon: &Syscon) {
+    // HSLSPI = High Speed Spi = Flexcomm 8
+    // The L stands for Let this just be named consistently for once
+    syscon.enable_clock(Peripheral::HsLspi).unwrap_lite();
+    syscon.leave_reset(Peripheral::HsLspi).unwrap_lite();
+
+    syscon.enable_clock(Peripheral::Fc3).unwrap_lite();
+    syscon.leave_reset(Peripheral::Fc3).unwrap_lite();
```

Comment:
Can go ahead and remove this since we're only using the HS-SPI

Reply:
Done